### PR TITLE
Improve handling of binder auto profile swapping.

### PR DESCRIPTION
**Pull request by aza547**
Hi,

I've been wanting this feature for a while. Decided to just have a go adding it, was very simple in the end. Maybe you want to incorporate it to the addon. 

Basically just makes the automatically Load_Profile on spec swap look for a profile called class-spec rather than just spec. Fixes the problem if you play two identically named specs like holy priest/pala or resto druid/shaman but want to have seperate keybindings for each. 

Thanks,
Alex

### CODE REVIEW

**Comment by JTensai** [comment]
Thanks for taking the time to create a pull request! Its great to have another pair of eyes look at this code :D

I'll merge this in if you can make it backwards compatible with people who are using the system as it already is.

Maybe run a check to see if there are any matching specs for class-specialization and if not, try to just load one for the specialization. That way people can be more specific if they want to.

Hope to see an update sometime!

**Review by JTensai — CHANGES_REQUESTED** [review]
See comment in conversation.

**Comment by aza547** [comment]
Made changes to be backwards compatible. Also includes a slight improvement to the existing behaviour - no longer attempts to load a profile that doesn't exist, rather than always trying to load a "spec" profile on respec event.

**Review by JTensai — APPROVED** [review]

**Comment by JTensai** [comment]
Looks great! Thanks for the update.

**Comment by aza547** [comment]
Very welcome!

**Pull request merged**
